web/src/views/InvestigationDetail.vue

```diff
@@ -498,17 +505,84 @@ export default defineComponent({
       }));
     }
 
+    async function toggleNonTiledImageOverlay(pin: Pin, dataset: Dataset) {
+      if (!dataset.content) return;
+      if (!featureLayer) return; // sanity check. this should exist
+      if (!nonTiledOverlayFeature) {
+        nonTiledOverlayFeature = featureLayer.createFeature('quad');
+      }
+      const quadData = nonTiledOverlayFeature.data() || [];
+      // eslint-disable-next-line
+      const existingOverlay = quadData.find((overlay: any) => overlay.pinId === pin.id);
+      if (existingOverlay) {
+        nonTiledOverlayFeature.data(
+          // eslint-disable-next-line
+          nonTiledOverlayFeature.data().filter((overlay: any) => overlay.pinId !== pin.id),
+        ).draw();
+        nonTiledOverlayFeature.draw();
+      } else {
+        try {
+          const urlRoot = process.env.VUE_APP_API_ROOT;
+          const url = `${urlRoot}/datasets/tile_source/${dataset.id}/thumbnail.png`;
+          const image: HTMLImageElement = await getNonTiledImage(url);
+          image.crossOrigin = 'use-credentials';
+          if (store.state.axiosInstance) {
```

Comment:
I would invert this test and place it at the top of the function, so you can early exit if `axiosInstance` doesn't exist yet, and you don't have to complexify the logic here with a type narrowing.

Reply:
Fixed in 2ec5d2c